Test/Machine/test_machine.py

```diff
@@ -82,11 +82,35 @@
 np.random.seed(12346)
 
 
+def same_derivatives(der_log, num_der_log):
+    assert(np.max(np.real(der_log - num_der_log))
+           == approx(0., rel=1e-4, abs=1e-4))
```

Comment:
Since we're using `double`s everywhere, is `1e-4` low enough?

Reply:
yeah it seems that we can't go much below 1.0e-6 though, I changed that. The thing is that these functions are typically highly non-linear and higher order terms can be very large in finite differences approximations